quickwit-metastore/src/split_metadata_version.rs

```diff
@@ -0,0 +1,215 @@
+    /// in the storage URI resolver: for instance, the Amazon S3 region.
+    pub split_id: String,
+
+    /// Number of records (or documents) in the split.
+    /// TODO make u64
+    #[serde(alias = "num_records")]
+    pub num_docs: usize,
+
+    /// Sum of the size (in bytes) of the documents in this split.
+    ///
+    /// Note this is not the split file size. It is the size of the original
+    /// JSON payloads.
+    pub size_in_bytes: u64,
+
+    /// If a timestamp field is available, the min / max timestamp in
+    /// the split.
+    pub time_range: Option<RangeInclusive<i64>>,
+
+    /// The state of the split. This is the only mutable attribute of the split.
+    pub split_state: SplitState,
+
+    /// Timestamp for tracking when the split was created.
+    #[serde(default = "utc_now_timestamp")]
+    pub create_timestamp: i64,
+
+    /// Timestamp for tracking when the split was last updated.
+    #[serde(default = "utc_now_timestamp")]
+    pub update_timestamp: i64,
+
+    /// A set of tags for categorizing and searching group of splits.
+    #[serde(default)]
+    pub tags: BTreeSet<String>,
+
+    /// Number of demux operations this split has undergone.
+    #[serde(default)]
+    pub demux_num_ops: usize,
+}
+
+#[derive(Clone, Eq, PartialEq, Default, Debug, Serialize, Deserialize)]
+pub(crate) struct SplitMetadataV1 {
+    /// Split ID. Joined with the index URI (<index URI>/<split ID>), this ID
+    /// should be enough to uniquely identify a split.
+    /// In reality, some information may be implicitly configured
+    /// in the storage URI resolver: for instance, the Amazon S3 region.
+    pub split_id: String,
+
+    /// Number of records (or documents) in the split.
+    pub num_docs: usize,
+
+    /// Sum of the size (in bytes) of the documents in this split.
+    ///
+    /// Note this is not the split file size. It is the size of the original
+    /// JSON payloads.
+    pub size_in_bytes: u64,
+
+    /// If a timestamp field is available, the min / max timestamp in
+    /// the split.
+    pub time_range: Option<RangeInclusive<i64>>,
+
+    /// Timestamp for tracking when the split was created.
+    #[serde(default = "utc_now_timestamp")]
+    pub create_timestamp: i64,
+
+    /// A set of tags for categorizing and searching group of splits.
+    #[serde(default)]
+    pub tags: BTreeSet<String>,
+
+    /// Number of demux operations this split has undergone.
+    #[serde(default)]
+    pub demux_num_ops: usize,
+
+    /// Contains the range of bytes of the footer that needs to be downloaded
+    /// in order to open a split.
+    ///
+    /// The footer offsets
+    /// make it possible to download the footer in a single call to `.get_slice(...)`.
+    pub footer_offsets: Range<u64>,
+}
+
+#[derive(Clone, Eq, PartialEq, Default, Debug, Serialize, Deserialize)]
+pub(crate) struct SplitMetadataAndFooterV0 {
+    split_metadata: SplitMetadataV0,
+    footer_offsets: Range<u64>,
+}
```

Comment:
Can you reorder the types in this file to have all V0 stuff in one place and after all of the V1 stuff.

Reply:
done